### ISSUE TITLE
Rocket Barrage does not stop firing during certain status conditions

### ISSUE THREAD

**Issue by TheZett** [ValveSoftware/Dota2-Gameplay#7087]
### Ability name

Rocket Barrage

### Description

According to an older update, Gyrocopter's Rocket Barrage is not supposed to fire its shots, if Gyrocopter is affected by the following status conditions:

- Banishment sources
- Hex sources
- Doom
- Chronosphere

Currently Gyrocopter does indeed stop firing his Rocket Barrage when affected by banishment sources (Disruption, Astral Imprisonment, etc), but the other 3 status conditions (hexes, doom, chrono) do not cause him to stop firing.

I've checked follow-up patches from March 2013 onwards and this was never officially reverted, hence me reporting this as a bug.

**Source:** https://dota2.fandom.com/wiki/March_07,_2013_Patch

### Example Match ID (and possibly Timestamp)

none

**Comment by shaileshgarg** [comment]
https://user-images.githubusercontent.com/26450759/221372791-635d7384-ed48-48f1-90e4-76a23e63e792.mp4


Doom does stop Rocket Barrage.

Rocket barrage is provided by a Buff on Gyro, and Hex/Chronoshpere don't dispel, so makes sense that Rocket barrage continues.

**Comment by TheZett** [comment]
Try it with the AoE doom, which does not dispel the target if it walks outside and back inside the AoE.

An active Rocket Barrage is not stopped when affected by the AoE Doom, despite the update stating that Rocket Barrage should stop firing when affected by any of these status effects.

**Comment by shaileshgarg** [comment]
Yeah, but the same point as above, Rocket Barrage is a Buff, and AOE doom doesn't dispel so it stays.

Would be weird if some abilities that don't dispel dispelled some abilities. But again Dota has many exceptions for whom bugs reports are open in this repository.

**Comment by TheZett** [comment]
Following your logic, banishment should not stop Rocket Barrage either, as banish spells do not dispel Gyrocopter either.

Since the patch said that Rocket Barrage is supposed to stop firing when getting affected by banishment sources, hexes, Doom or Chrono, I reported it as a bug.

Rocket Barrage is not dispelled, but its effect are supposed to be temporarily stopped when these status conditions are detected.

**Comment by cadaeic** [comment]
Might just as well be an unannounced intentional change. In particular Hex and Doom stopping Rocket Barrage felt very "Dota 1" and needlessly convoluted.
